### PR TITLE
[DB-603] [risk=low] Generate ehr concept counts by considering ehr records

### PR DESCRIPTION
**Pull request by SrushtiGangireddy**
1. Create views from mapping tables (if present).
2. Use these views to generate counts for all the concepts.
3. Use the full tables to generate counts for only physical measurements.

### CODE REVIEW

File: public-api/db-cdr/generate-cdr/run-achilles-queries.sh [hunk 1]
```diff
@@ -46,6 +46,80 @@ fi
...
+    echo "CREATE VIEWS - v_ehr_drug_exposure"
+    bq --quiet --project=$BQ_PROJECT query --nouse_legacy_sql \
+    "CREATE OR REPLACE VIEW \`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.v_ehr_drug_exposure\` AS
+    select m.* from \`${BQ_PROJECT}.${BQ_DATASET}.drug_exposure\` m join \`${BQ_PROJECT}.${BQ_DATASET}._mapping_drug_exposure\` mm
+    on m.drug_exposure_id = mm.drug_exposure_id
+    where mm.src_dataset_id=(select distinct src_dataset_id from \`${BQ_PROJECT}.${BQ_DATASET}._mapping_drug_exposure\` where src_dataset_id like '%ehr%')
+    and (m.drug_concept_id > 0 or m.drug_source_concept_id > 0)"
+
+else
+    echo "CREATE VIEWS - v_ehr_condition_occurrence"
```
Comment (calbach) [inline public-api/db-cdr/generate-cdr/run-achilles-queries.sh:98]: measurement

File: public-api/db-cdr/generate-cdr/run-achilles-queries.sh [hunk 2]
```diff
@@ -46,6 +46,80 @@ fi
...
 tables=$(bq --project=$BQ_PROJECT --dataset=$BQ_DATASET ls)
 
+################################################
+# CREATE VIEWS
+################################################
+
+# We want to fetch the breakdown counts of each analysis for all of the ehr domain concepts by considering only the records that are ehr related (no ppi results are such)
+# So, instead of joining each of the query with the mapping tables we create the view which are used for ehr only count queries.
+# Mapping tables have the dataset id specified which can be used to differentiate ehr specific rows
+if [[ "$tables" == *"_mapping_"* ]]; then
```
Comment (calbach) [inline public-api/db-cdr/generate-cdr/run-achilles-queries.sh:56]: In what situations would this not be here? Is this a synthetic vs prod thing?

File: public-api/db-cdr/generate-cdr/run-achilles-queries.sh [hunk 3]
```diff
@@ -1504,187 +1577,38 @@ where (o.observation_source_concept_id > 0 and o.value_source_concept_id = 0 and
...
-    (select procedure_source_concept_id as concept,co.person_id as person
-    from \`${BQ_PROJECT}.${BQ_DATASET}.procedure_occurrence\` co join \`${BQ_PROJECT}.${BQ_DATASET}.concept\` c on co.procedure_source_concept_id=c.concept_id
-    where c.vocabulary_id != 'PPI' and c.concept_id not in (3036277,903133,903118,903115,3025315,903121,903135,903136,903126,903111,42528957,903120)
-    and co.procedure_source_concept_id not in (select distinct procedure_concept_id from \`${BQ_PROJECT}.${BQ_DATASET}.procedure_occurrence\`)),
-    concepts as
-    (select * from procedure_concepts union all select * from procedure_source_concepts)
-    select 0 as id,3000 as analysis_id,'10' as stratum_1,'Procedure' as stratum_3, (select count(distinct person) from concepts) as count_value,
-    0 as source_count_value"
-fi
+### Test data does not have the mapping tables, so this else block lets the script to fetch domain counts for test data
```
Comment (calbach) [inline public-api/db-cdr/generate-cdr/run-achilles-queries.sh:1580]: Ah, this is the comment I was looking for above, should this be moved as well?

File: public-api/db-cdr/generate-cdr/run-measurement-queries.sh [hunk 4]
```diff
@@ -44,27 +44,610 @@ then
...
+
 # Next Populate achilles_results
 echo "Running measurement queries..."
 
+# Run these queries in prod to generate counts for ppi concepts from the full measurement table (and not just the ehr one used to generate ehr counts for rest of the concepts)
+# In case of test data, the views has all the data, so we do not need to run these.
+# If mapping tables are present, the views contain ehr specific data, so we need to run these extra queries.
+if [[ "$tables" == *"_mapping_"* ]]; then
+    # 3000 Measurements that have numeric values - Number of persons with at least one measurement occurrence by measurement_concept_id, bin size of the measurement value for 10 bins, maximum and minimum from measurement value. Added value for measurement rows
+    bq --quiet --project=$BQ_PROJECT query --nouse_legacy_sql \
```
Comment (calbach) [inline public-api/db-cdr/generate-cdr/run-measurement-queries.sh:58]: Where did all of this new code come from? Was this a copy paste from somewhere?

File: public-api/db-cdr/generate-cdr/run-achilles-queries.sh (hunk 1 above)
Comment (SrushtiGangireddy) [reply public-api/db-cdr/generate-cdr/run-achilles-queries.sh:98]: Done. Thanks.

File: public-api/db-cdr/generate-cdr/run-achilles-queries.sh (hunk 2 above)
Comment (SrushtiGangireddy) [reply public-api/db-cdr/generate-cdr/run-achilles-queries.sh:56]: yes exactly, the test/ synthetic data does not have any of the mapping tables.

File: public-api/db-cdr/generate-cdr/run-achilles-queries.sh (hunk 3 above)
Comment (SrushtiGangireddy) [reply public-api/db-cdr/generate-cdr/run-achilles-queries.sh:1580]: Moved the comment to the top of if block.

File: public-api/db-cdr/generate-cdr/run-measurement-queries.sh (hunk 4 above)
Comment (SrushtiGangireddy) [reply public-api/db-cdr/generate-cdr/run-measurement-queries.sh:58]: this queries are exactly run below for rest of the ehr measurements considering only the ehr record rows in v_ehr_measurement. I wanted these to run on the full measurement table to calculate the counts of just the physical measurements. (which do not have counts if these queries are not run)

**Review by calbach — APPROVED** [review]

**Pull request merged**